api/CcsSso.Core.Service/External/OrganisationProfileService.cs

```diff
@@ -723,13 +723,21 @@ public string GetCountryNameByCode(string countyCode)
     private async Task<List<OrganisationEligibleRole>> GetOrganisationEligibleRolesAsync(Organisation org, int supplierBuyerType)
     {
       var eligibleRoles = new List<OrganisationEligibleRole>();
-      if (supplierBuyerType == 0)
+      if (supplierBuyerType == 0) //Supplier
       {
         var roles = await _dataContext.CcsAccessRole.Where(ar => !ar.IsDeleted &&
           ar.SubscriptionTypeEligibility == RoleEligibleSubscriptionType.Default &&
           ar.OrgTypeEligibility != RoleEligibleOrgType.Internal &&
           (ar.TradeEligibility == RoleEligibleTradeType.Supplier || ar.TradeEligibility == RoleEligibleTradeType.Both)
         ).ToListAsync();
+
+        var defroles = await _dataContext.CcsAccessRole.Where(ar => !ar.IsDeleted &&
```

Comment:
You dont need to do any code change. Just try to change the eligibility values for these roles.

Reply:
done the changes as per your comment. please review do not merge.